src/apply_rules.rs

```diff
@@ -82,6 +96,91 @@ fn read_complementary_code_rules_files<P: AsRef<Path>>(
 }
 
+fn read_property_rules_files<P: AsRef<Path>>(
+    rule_files: Vec<P>,
+    report: &mut Report,
+) -> Result<Vec<PropertyRule>> {
+    info!("Reading property rules.");
+    let mut properties: BTreeMap<(ObjectType, String, String), BTreeSet<PropertyRule>> =
+        BTreeMap::default();
+    for rule_path in rule_files {
+        let path = rule_path.as_ref();
+        let mut rdr = csv::ReaderBuilder::new()
+            .trim(csv::Trim::All)
+            .from_path(&path)
+            .with_context(ctx_from_path!(path))?;
+        for p in rdr.deserialize() {
+            let p: PropertyRule = match p {
+                Ok(val) => val,
+                Err(e) => {
+                    report.add_warning(
+                        format!("Error reading {:?}: {}", path.file_name().unwrap(), e),
+                        ReportType::PropertyRulesRead,
+                    );
+                    continue;
+                }
+            };
+
+            properties
+                .entry((p.object_type, p.object_id.clone(), p.property_name.clone()))
+                .or_insert_with(BTreeSet::new)
+                .insert(p);
+        }
+    }
+
+    let properties = properties
+        .iter()
+        .filter(|((object_type, object_id, property_name), property)| {
+            match object_type {
+                ObjectType::Route
+                    if ![
+                        "route_name",
+                        "direction_type",
+                        "route_geometry",
+                        "destination_id",
+                    ]
+                    .contains(&property_name.as_ref()) =>
+                {
+                    report.add_warning(
+                        format!(
+                            "object_type={}, object_id={}: unknown property_name {} defined",
+                            object_type.as_str(), object_id, property_name,
+                        ),
+                        ReportType::UnknownPropertyName,
+                    );
+                    return false;
+                }
+                ObjectType::Line | ObjectType::StopPoint | ObjectType::StopArea => {
+                    warn!(
+                        "Changing properties for {:?} is not yet possible.",
+                        object_type.as_str()
+                    );
+                    return false;
+                }
+                _ => {}
+            }
+
+            if property.len() > 1 {
+                {
+                    report.add_warning(
+                        format!(
+                            "object_type={}, object_id={}: multiple values specified for the property {}",
+                            object_type.as_str(), object_id, property_name
+                        ),
+                        ReportType::MultipleValue,
+                    );
+                }
+                return false;
+            }
+            true
+        })
+        .flat_map(|(_, p)| p)
+        .cloned()
```

Comment:
it's needed because
```
a collection of type `std::vec::Vec<apply_rules::PropertyRule>` cannot be built from `std::iter::Iterator<Item=&apply_rules::PropertyRule>
```

Reply:
I think this can be resolved by using `into_iter()` instead of `iter()`

Reply:
:+1: